src/Microsoft.TemplateSearch.Common/FileMetadataTemplateSearchCache.cs

```diff
@@ -55,8 +55,10 @@ public IReadOnlyList<ITemplateInfo> GetNameMatchedTemplates(string searchName)
                 return _templateDiscoveryMetadata.TemplateCache;
             }
 
-            return _templateDiscoveryMetadata.TemplateCache.Where(template => template.Name.IndexOf(searchName, StringComparison.OrdinalIgnoreCase) >= 0
-                                                    || template.ShortName.IndexOf(searchName, StringComparison.OrdinalIgnoreCase) >= 0).ToList();
+            return _templateDiscoveryMetadata.TemplateCache.Where(
+                template => template.Name.IndexOf(searchName, StringComparison.OrdinalIgnoreCase) >= 0
+                || template.ShortNameList.Any(shortName => shortName.IndexOf(searchName, StringComparison.OrdinalIgnoreCase) >= 0))
```

Comment:
Not your change: Can we replace `string.IndexOf` here with `string.Contains`?

Reply:
We need case insensitive here.